contracts/src/v0.8/ccip/DonIdClaimer.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {ICapabilityConfiguration} from "../keystone/interfaces/ICapabilityConfiguration.sol";
+
+contract DonIDClaimer {
+  error ZeroAddressNotAllowed();
+
+  uint256 public donID;  
+  address internal immutable i_capabilitiesRegistry;
+
+  constructor(
+    address capabilitiesRegistry
+  ) {
+    if (capabilitiesRegistry == address(0)) {
+      revert ZeroAddressNotAllowed();
+    }
+    i_capabilitiesRegistry = capabilitiesRegistry;
+  }
+
+  function claimNextDonID() external returns (uint256) {
+    donID = donID + 1;
+    return donID; 
+  }
+
+  function setDONID(uint256 donId) external pure returns (uint256) {
+      return donId; 
+  }
+
+  function getDonID() external view returns (uint256) {
+      return donID; 
+  }
+  
+  function syncDonIdWithCapReg() external {
```

Comment:
Can we add an offset parameter to the input? I can see scenarios where we need to sync, but add an offset to account for some proposals that will still be executed.

Reply:
(Approval was accidental)

Reply:
https://github.com/smartcontractkit/chainlink/pull/16926/commits/e5e4a9ed063b98f455d7792e02a2adf5da36ca33